test/e2e/apps/job.go

```diff
@@ -817,6 +818,58 @@ var _ = SIGDescribe("Job", func() {
 		gomega.Expect(jobs.Items).To(gomega.BeEmpty(), "Found job %v", jobName)
 	})
 
+	ginkgo.It("should check the ready of a pods", func(ctx context.Context) {
+		backoff := 0
+		ginkgo.By("Creating a job")
+		job := e2ejob.NewTestJob("notTerminate", "all-ready", v1.RestartPolicyNever, parallelism, completions, nil, int32(backoff))
+
+		job, err := e2ejob.CreateJob(ctx, f.ClientSet, f.Namespace.Name, job)
+		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
+
+		ginkgo.By("Ensuring job has ready pods")
+		err = e2ejob.WaitForJobReady(ctx, f.ClientSet, f.Namespace.Name, job.Name, ready)
+		framework.ExpectNoError(err, "failed to ensure job ready in namespace: %s", f.Namespace.Name)
+
+		ginkgo.By("Ensuring pods for job exist")
+		pods, err := e2ejob.GetJobPods(ctx, f.ClientSet, f.Namespace.Name, job.Name)
+		framework.ExpectNoError(err, "failed to get pod list for job in namespace: %s", f.Namespace.Name)
+		podsReady := int32(0)
```

Comment:
Don't need to verify the pod creation, and pods ready field. This is set by kubetlet, so should be already tested in other tests.